windows/lcow_sandbox.go

```diff
@@ -0,0 +1,521 @@
+	if err := svm.config.StartUtilityVM(); err != nil {
+		return nil, fmt.Errorf("failed to start service utility VM (%s): %s", context, err)
+	}
+
+	// As it's now running, add it to the map, checking for a race where another
+	// thread has simultaneously tried to start it.
+	log.G(ctx).Debugf("%s locking serviceVmsMutex for insertion", title)
+	r.serviceVmsMutex.Lock()
+	if svm, ok := r.serviceVms[id]; ok {
+		log.G(ctx).Debugf("%s releasing serviceVmsMutex after insertion but exists", title)
+		r.serviceVmsMutex.Unlock()
+		return svm, nil
+	}
+	r.serviceVms[id] = svm
+	log.G(ctx).Debugf("%s releasing serviceVmsMutex after insertion", title)
+	r.serviceVmsMutex.Unlock()
+
+	// Now we have a running service VM, we can create the cached scratch file if it doesn't exist.
+	log.G(ctx).Debugf("%s locking cachedScratchMutex", title)
+	r.cachedScratchMutex.Lock()
+	if _, err := os.Stat(r.cachedScratchFile); err != nil {
+		log.G(ctx).Debugf("%s (%s): creating an SVM scratch - locking serviceVM", title, context)
+		svm.Lock()
+		if err := svm.config.CreateExt4Vhdx(scratchTargetFile, client.DefaultVhdxSizeGB, r.cachedScratchFile); err != nil {
+			log.G(ctx).Debugf("%s (%s): releasing serviceVM on error path from CreateExt4Vhdx: %s", title, context, err)
+			svm.Unlock()
+			log.G(ctx).Debugf("%s (%s): releasing cachedScratchMutex on error path", title, context)
+			r.cachedScratchMutex.Unlock()
+
+			// Do a force terminate and remove it from the map on failure, ignoring any errors
+			if err2 := r.terminateServiceVM(ctx, id, "error path from CreateExt4Vhdx", true); err2 != nil {
+				log.G(ctx).Warnf("failed to terminate service VM on error path from CreateExt4Vhdx: %s", err2)
+			}
+
+			return nil, fmt.Errorf("failed to create SVM scratch VHDX (%s): %s", context, err)
+		}
+		log.G(ctx).Debugf("%s (%s): releasing serviceVM after %s created and cached to %s", title, context, scratchTargetFile, r.cachedScratchFile)
+		svm.Unlock()
+	}
+	log.G(ctx).Debugf("%s (%s): releasing cachedScratchMutex", title, context)
+	r.cachedScratchMutex.Unlock()
+
+	// Hot-add the scratch-space if not already attached
+	if !svm.scratchAttached {
+		log.G(ctx).Debugf("lcowdriver: startServiceVmIfNotRunning: (%s) hot-adding scratch %s - locking serviceVM", context, scratchTargetFile)
+		svm.Lock()
+		if err := svm.config.HotAddVhd(scratchTargetFile, toolsScratchPath, false, true); err != nil {
+			log.G(ctx).Debugf("%s (%s): releasing serviceVM on error path of HotAddVhd: %s", title, context, err)
+			svm.Unlock()
+
+			// Do a force terminate and remove it from the map on failure, ignoring any errors
+			if err2 := r.terminateServiceVM(ctx, id, "error path from HotAddVhd", true); err2 != nil {
+				log.G(ctx).Warnf("failed to terminate service VM on error path from HotAddVhd: %s", err2)
+			}
+
+			return nil, fmt.Errorf("failed to hot-add %s failed: %s", scratchTargetFile, err)
+		}
+		log.G(ctx).Debugf("%s (%s): releasing serviceVM", title, context)
+		svm.Unlock()
```

Comment:
shouldn't `svm.scratchAttached` be set to `true` here?

Reply:
yep